roles/common/tasks/monitoring.yml

```diff
@@ -58,12 +58,65 @@
   lineinfile: dest=/etc/default/sensu regexp=^EMBEDDED_RUBY
               line=EMBEDDED_RUBY=true
   notify: restart sensu-client
+  
+- block:
+  - name: sensu python venv directory
+    file: dest="{{ common.python.sensu_venv }}"
+          state=directory mode=0755
+        
+  - name: set pip proxy args
+    set_fact:
+      pip_extra_args: " -i {{ openstack.pypi_mirror }}"
+    when: openstack.pypi_mirror is defined
+
+  - name: update pip
+    pip: name=pip version={{ common.pip_version }}
+         extra_args="-i {{ openstack.pypi_mirror }}"
+         virtualenv="{{ common.python.sensu_venv }}"
+    register: result
+    until: result|succeeded
+    retries: 5
+
+  - name: upgrade setuptools for sensu venv (osp)
+    pip:
+      name: "setuptools"
+      virtualenv: "{{ common.python.sensu_venv }}"
+      extra_args: "{{ pip_extra_args }} --upgrade"
+    register: result
+    until: result|succeeded
+    retries: 5
+
+  - name: install shade bits for ansible modules
+    pip:
+      name: "{{ item }}"
+      virtualenv: "{{ common.python.sensu_venv }}"
+    with_items:
+      - shade>=1.9.0
+      - python-dateutil
+    register: result
+    until: result|succeeded
+    retries: 5
+
+  - name: create symbolic link for ca cert
+    file: 
+      src: "{{ ca_bundle }}"
+      dest: "{{ common.python.sensu_venv }}/lib/python2.7/site-packages/requests/cacert.pem"
+      state: link
+      force: true
+
+  when: openstack_install_method == "distro"
 
 - name: ensure PATH contains /usr/local/bin (OpenStack clients)
   lineinfile: dest=/etc/default/sensu regexp=^PATH
               line="PATH=/opt/sensu/embedded/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/etc/sensu/plugins"
   notify: restart sensu-client
 
+- name: ensure sensu client use sensu_venv python
+  lineinfile: dest=/etc/default/sensu regexp="^\. {{ common.python.sensu_venv }}/bin/activate"
```

Comment:
feels like there should be abetter way to handle this?  can we set the approate env variable so that `#!/usr/bin/env python` resolves to our virtualenv python?  I think that's all that would be needed.